### PR TITLE
PATCH: Bumped up PythonConnector PATCH version from 3.12.1 to 3.12.2

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
@noreview - This is an automated process. No review is required

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

4. (Optional) PR for stored-proc connector:

### CODE REVIEW

File: DESCRIPTION.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
-- v3.12.2(TBD)
-  - Enhanced error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
-  - Improved implementation of `snowflake.connector.util_text.random_string` to avoid collisions.
-  - If the account specifies a region, and that region is in China, the TLD is now inferred to be snowflakecomputing.cn.
-  - Changed loglevel to WARNING for OCSP fail-open warning messages (was: ERROR)
+- v3.12.2(September 10,2024)
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:11]:
```suggestion
- v3.12.2(September 11,2024)
```

File: DESCRIPTION.md [hunk 2]
```diff
@@ -8,11 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
 
-- v3.12.2(TBD)
-  - Enhanced error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
-  - Improved implementation of `snowflake.connector.util_text.random_string` to avoid collisions.
-  - If the account specifies a region, and that region is in China, the TLD is now inferred to be snowflakecomputing.cn.
-  - Changed loglevel to WARNING for OCSP fail-open warning messages (was: ERROR)
+- v3.12.2(September 11,2024)
+  - Improved error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
+  - Improved inference of TLD for accounts specifying a region in China, now defaulting to snowflakecomputing.cn.
+  - Improved implementation of snowflake.connector.util_text.random_string to reduce the likelihood of collisions.
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:14]: of the

File: DESCRIPTION.md [hunk 3]
```diff
@@ -8,11 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
 # Release Notes
 
-- v3.12.2(TBD)
-  - Enhanced error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
-  - Improved implementation of `snowflake.connector.util_text.random_string` to avoid collisions.
-  - If the account specifies a region, and that region is in China, the TLD is now inferred to be snowflakecomputing.cn.
-  - Changed loglevel to WARNING for OCSP fail-open warning messages (was: ERROR)
+- v3.12.2(September 11,2024)
+  - Improved error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
+  - Improved inference of TLD for accounts specifying a region in China, now defaulting to snowflakecomputing.cn.
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:13]: is TLD a well-understood thing?

File: DESCRIPTION.md [hunk 4]
```diff
@@ -8,11 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
+  - Improved implementation of snowflake.connector.util_text.random_string to reduce the likelihood of collisions.
+  - Updated log level for OCSP fail-open warning messages from ERROR to WARNING.
```
Comment (sfc-gh-mheavin) [inline DESCRIPTION.md:15]: Update the log...

**Review by sfc-gh-mheavin — COMMENTED** [review]
Some thoughts for your consideration

File: DESCRIPTION.md (hunk 4 above)
Comment (sfc-gh-aling) [reply DESCRIPTION.md]: updated

File: DESCRIPTION.md (hunk 2 above)
Comment (sfc-gh-aling) [reply DESCRIPTION.md]: updated

File: DESCRIPTION.md (hunk 3 above)
Comment (sfc-gh-aling) [reply DESCRIPTION.md]: I debreivated it to be "top level domain"

Comment (sfc-gh-mheavin) [reply DESCRIPTION.md]: In that case, grammatically, it should be top level domains

File: DESCRIPTION.md [hunk 5]
```diff
@@ -8,11 +8,11 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
 # Release Notes
 
-- v3.12.2(TBD)
-  - Enhanced error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
-  - Improved implementation of `snowflake.connector.util_text.random_string` to avoid collisions.
-  - If the account specifies a region, and that region is in China, the TLD is now inferred to be snowflakecomputing.cn.
-  - Changed loglevel to WARNING for OCSP fail-open warning messages (was: ERROR)
+- v3.12.2(September 11,2024)
+  - Improved error handling for asynchronous queries, providing more detailed and informative error messages when an async query fails.
+  - Improved inference of top level domain for accounts specifying a region in China, now defaulting to snowflakecomputing.cn.
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:13]:
```suggestion
  - Improved inference of top level domains for accounts specifying a region in China, now defaulting to snowflakecomputing.cn.
```

Comment (sfc-gh-mheavin) [inline DESCRIPTION.md]: top-level domains

**Review by sfc-gh-mheavin — APPROVED** [review]
One minor

File: DESCRIPTION.md (hunk 5 above)
Comment (sfc-gh-aling) [reply DESCRIPTION.md]: updated

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**